templates/cluster-template-external-cloud-provider.yaml

```diff
@@ -3,7 +3,6 @@ apiVersion: cluster.x-k8s.io/v1alpha4
 kind: Cluster
 metadata:
   name: ${CLUSTER_NAME}
-  namespace: '${NAMESPACE}'
```

Comment:
This was added with https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/735
@sbueringer I hope it does not conflict with it.

Reply:
Yes, CAPO works without this. Only I followed CAPI recommendation
https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#common-variables

I thought someone uses multi cloud provider and they wonder they can not use `--target-namespace` for CAPO.

But CAPI recommends this 
https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#common-variables.
I followed

Reply:
I think it was okay that we had it before. My interpretation of the documentation is (now):

* If you have places in you're YAML were you need the target namespace, you can/should use this variable. It is not necessary to use it for the metadata.namespace field as that is set by CAPI in any case

I looked at CAPG / CAPA / CAPZ / CAPV and I only could find it in CAPV. As it's not necessary and most of the other providers are not using it I thought it's cleaner to drop it.